runtime/doc/api.txt

```diff
@@ -3152,129 +3152,203 @@ nvim_tabpage_set_var({tabpage}, {name}, {value})
+                autocommands).
 
-                Parameters: ~
-                    {event}  The event or events to register this autocmd
-                             Required keys: event: string | ArrayOf(string)
+                Example using callback: >
+                    -- Lua function
+                    local myluafun = function() print("This buffer enters") end
 
-                Parameters: ~
-                    {opts}  Optional Parameters:
-                            • callback: (string|function)
-                              • (string): The name of the viml function to
-                                execute when triggering this autocmd
-                              • (function): The lua function to execute when
-                                triggering this autocmd
-                              • NOTE: Cannot be used with {command}
+                    -- Vimscript function name (as a string)
+                    local myvimfun = "g:MyVimFunction"
 
-                            • command: (string) command
-                              • vimscript command
-                              • NOTE: Cannot be used with {callback} Eg.
-                                command = "let g:value_set = v:true"
+                    vim.api.nvim_create_autocmd({
+                      event = {"BufEnter", "BufWinEnter"},
+                      pattern = {"*.c", "*.h"},
+                      callback = myluafun,  -- Or myvimfun
+                    })
+<
+
+                Example using command: >
+                    vim.api.nvim_create_autocmd({
+                      event = {"BufEnter", "BufWinEnter"},
+                      pattern = {"*.c", "*.h"},
+                      command = "echo 'Entering a C or C++ file'",
+                    })
+<
 
-                            • pattern: (string|table)
-                              • pattern or patterns to match against
-                              • defaults to "*".
-                              • NOTE: Cannot be used with {buffer}
+                Example values for pattern: >
+                  pattern = "*.py"
+                  pattern = { "*.py", "*.pyi" }
+<
 
-                            • buffer: (bufnr)
-                              • create a |autocmd-buflocal| autocmd.
-                              • NOTE: Cannot be used with {pattern}
+                Examples values for event: >
+                  event = "BufPreWrite"
+                  event = {"CursorHold", "BufPreWrite", "BufPostWrite"}
+<
 
-                            • group: (string|int) The augroup name or id
-                            • once: (boolean) - See |autocmd-once|
-                            • nested: (boolean) - See |autocmd-nested|
-                            • desc: (string) - Description of the autocmd
+                Parameters: ~
+                    {event}  (String|Array) The event or events to register
+                             this autocmd
```

Comment:
```suggestion
                             this autocommand
```
(and similar for the `{opts}` description).

Big 👍🏻 on standardizing these, and opting for the full term!